elasticapm/traces.py

```diff
@@ -709,13 +740,55 @@ def set_transaction_name(name, override=True):
 
 
 def set_transaction_result(result, override=True):
+    """
+    Sets the result of the transaction. The result could be e.g. the HTTP status class (e.g "HTTP 5xx") for
+    HTTP requests, or "success"/"fail" for background tasks.
+
+    :param name: the name of the transaction
+    :param override: if set to False, the name is only set if no name has been set before
+    :return: None
+    """
+
     transaction = execution_context.get_transaction()
     if not transaction:
         return
     if transaction.result is None or override:
         transaction.result = result
 
 
+def set_transaction_outcome(outcome=None, http_status_code=None, override=True):
```

Comment:
I'm not sure whether we need the override parameter in the public API.
What would be a use case where you wouldn't want to override?
Can there ever be a situation where the agent already set the outcome when the user calls this method?

Reply:
It's always possible for people to provide framework integrations as external modules, and for those cases it would be useful to have this I think.

Generally, we don't really have a different internal API. What makes an API public (in the sense that we commit to API stability) is the documentation. What we do not document, we consider private, and if enterprising users find the APIs by reading code, they will have to expect breaking changes at any time.